languages/thingtalk/load-thingpedia.js

```diff
@@ -31,7 +31,9 @@ class ThingpediaLoader {
         this._grammar = grammar;
 
         this._tpClient = options.thingpediaClient;
-        this._schemas = options.schemaRetriever || new SchemaRetriever(this._tpClient, null, !options.debug);
+        if (!options.schemaRetriever)
+            options.schemaRetriever = new SchemaRetriever(this._tpClient, null, !options.debug);
+        this._schemas = options.schemaRetriever;
```

Comment:
why this change?

Reply:
Before this PR, schemaRetriever was usually provided by the API caller (ie tool/generate.js, or almond-cloud). Now it is not, because we want to cut the number of files that `require('thingtalk')`. But we still don't want to create too many schema retrievers, or we lose the caching they provide and hit the Thingpedia API too often.
So with this change, if the schema retriever is created lazily, it is also stored in `options` for other code to use (specifically, for languages/thingtalk.js), and we only create it once.